src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginGlyph.xaml.cs

```diff
@@ -50,53 +50,85 @@ internal partial class InheritanceMarginGlyph
             _operationExecutor = operationExecutor;
             _textView = textView;
             _listener = listener;
-            InitializeComponent();
+
+            Background = Brushes.Transparent;
+            BorderBrush = Brushes.Transparent;
+
+            Click += InheritanceMargin_OnClick;
+            MouseEnter += InheritanceMargin_OnMouseEnter;
+            MouseLeave += InheritanceMargin_OnMouseLeave;
+
+            Resources.Add(ToolTipStyleKey, new Style(typeof(ToolTip))
+            {
+                Setters =
+                {
+                    new Setter(BackgroundProperty, new DynamicResourceExtension(EnvironmentColors.ToolTipBrushKey)),
+                    new Setter(BorderBrushProperty, new DynamicResourceExtension(EnvironmentColors.ToolTipBorderBrushKey)),
+                    new Setter(ForegroundProperty, new DynamicResourceExtension(EnvironmentColors.ToolTipTextBrushKey)),
+                },
+            });
 
             var viewModel = InheritanceMarginGlyphViewModel.Create(classificationTypeMap, classificationFormatMap, tag, textView.ZoomLevel);
+            SetValue(AutomationProperties.NameProperty, viewModel.AutomationName);
+
+            // Control template only shows the image
+            var templateBorder = new FrameworkElementFactory(typeof(Border), "Border");
+            templateBorder.SetValue(Border.BorderThicknessProperty, new Thickness(1));
+            templateBorder.SetValue(Border.BackgroundProperty, new TemplateBindingExtension(BackgroundProperty));
+            templateBorder.SetValue(Border.BorderBrushProperty, new TemplateBindingExtension(BorderBrushProperty));
+
+            var templateImage = new FrameworkElementFactory(typeof(CrispImage));
+            templateImage.SetValue(CrispImage.MonikerProperty, viewModel.ImageMoniker);
+            templateImage.SetValue(CrispImage.ScaleFactorProperty, viewModel.ScaleFactor);
+            templateBorder.AppendChild(templateImage);
+
+            Template = new ControlTemplate { VisualTree = templateBorder };
             DataContext = viewModel;
-            ContextMenu.DataContext = viewModel;
-            ToolTip = new ToolTip { Content = viewModel.ToolTipTextBlock, Style = (Style)FindResource("ToolTipStyle") };
         }
 
-        private void InheritanceMargin_OnClick(object sender, RoutedEventArgs e)
+        protected override void OnMouseEnter(MouseEventArgs e)
         {
-            if (this.ContextMenu != null)
-            {
-                this.ContextMenu.IsOpen = true;
-                e.Handled = true;
-            }
+            LazyInitialize();
```

Comment:
😢 I really don't like these random calls to LazyInitialize(), but I don't know a clean way to lazily-construct a context menu.

Reply:
Am trying this alternative
https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/how-to-handle-the-contextmenuopening-event?view=netframeworkdesktop-4.8

Reply:
➡️ OK this is way better now.